lms/djangoapps/course_home_api/progress/v1/views.py

```diff
@@ -0,0 +1,107 @@
+        course_blocks:
+            blocks: List of serialized Course Block objects. Each serialization has the following fields:
+                id: (str) The usage ID of the block.
+                type: (str) The type of block. Possible values the names of any
+                    XBlock type in the system, including custom blocks. Examples are
+                    course, chapter, sequential, vertical, html, problem, video, and
+                    discussion.
+                display_name: (str) The display name of the block.
+                lms_web_url: (str) The URL to the navigational container of the
+                    xBlock on the web LMS.
+                children: (list) If the block has child blocks, a list of IDs of
+                    the child blocks.
+        enrollment_mode: (str) a str representing the enrollment the user has ('audit', 'verified', ...)
+    **Returns**
+
+        * 200 on success with above fields.
+        * 403 if the user is not authenticated.
+        * 404 if the course is not available or cannot be seen.
+    """
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = ProgressTabSerializer
+
+    def get(self, request, *args, **kwargs):
+        course_key_string = kwargs.get('course_key_string')
+        course_key = CourseKey.from_string(course_key_string)
+        course_usage_key = modulestore().make_course_usage_key(course_key)
+
+        # Enable NR tracing for this view based on course
+        monitoring_utils.set_custom_metric('course_id', course_key_string)
+        monitoring_utils.set_custom_metric('user_id', request.user.id)
+        monitoring_utils.set_custom_metric('is_staff', request.user.is_staff)
+
+        _, request.user = setup_masquerade(
+            request,
+            course_key,
+            staff_access=has_access(request.user, 'staff', course_key),
+            reset_masquerade_data=True
+        )
+
+        transformers = BlockStructureTransformers()
+        transformers += course_blocks_api.get_course_block_access_transformers(request.user)
+        transformers += [
+            BlocksAPITransformer(None, None, depth=3),
+        ]
+        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=True)
+        course_blocks = get_course_blocks(request.user, course_usage_key, transformers, include_completion=True)
+
+        enrollment_mode, _ = CourseEnrollment.enrollment_mode_for_user(request.user, course_key)
+
+        course_grade = CourseGradeFactory().read(request.user, course)
+        courseware_summary = list(course_grade.chapter_grades.values())
+
+        data = {
+            'course_blocks': course_blocks,
+            'enrollment_mode': enrollment_mode,
+            'courseware_summary': courseware_summary,
+        }
+        context = self.get_serializer_context()
+        context['staff_access'] = request.user.is_staff
```

Comment:
Hold up. This is very easy to get twisted, because we named things badly.

`user.is_staff` is a Django concept (consider it like, edX staff). But in the context of a course, a person with staff access is both edX staff and instructors and TAs and some others (that is what `has_access(request.user, 'staff', course_key)` above is checking for).

You might be wanting that version of staff here rather than `user.is_staff`, if this is controlling whether they see grades.